controllers/schedule_controller.go

```diff
@@ -41,18 +45,50 @@ func (r *ScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return reconcile.Result{}, err
 	}
 
+	effectiveSchedule, err := r.FetchEffectiveScheduleResource(ctx, schedule)
+	if err != nil {
+		r.Log.Info("could not retrieve list of effective schedules, try later again", "error", err.Error())
+		return ctrl.Result{Requeue: true, RequeueAfter: 60 * time.Minute}, err
+	}
+
 	repository := cfg.GetGlobalRepository()
 	if schedule.Spec.Backend != nil {
 		repository = schedule.Spec.Backend.String()
 	}
 	config := job.NewConfig(ctx, r.Client, log, schedule, r.Scheme, repository)
 
-	return ctrl.Result{}, handler.NewScheduleHandler(config, schedule).Handle()
+	return ctrl.Result{}, handler.NewScheduleHandler(config, schedule, effectiveSchedule).Handle()
 }
 
+// SetupWithManager configures the reconciler.
 func (r *ScheduleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&k8upv1alpha1.Schedule{}).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// FetchEffectiveScheduleResource retrieves a list of EffectiveSchedules and filter the one that matches the given schedule.
+// Returns an error if the listing failed, but nil when no matching EffectiveSchedule object was found.
+func (r *ScheduleReconciler) FetchEffectiveScheduleResource(ctx context.Context, schedule *k8upv1alpha1.Schedule) (*k8upv1alpha1.EffectiveSchedule, error) {
```

Comment:
I wonder if it makes more sense to have a bidirectional link, e.g. each Schedule object also maintains a link to the EffectiveSchedule with name, namespace and jobtype info. That way, we won't be needing to fetch and filter all effective Schedules, possibly lowering the load on the API a bit. However, bidirectional links are easier to break.

WDYT ?

Reply:
I wouldn't go for the two-way binding. It only helps us very little but IMO adds additional complexity. Also we'd have to think about what happens if they get out of sync. Also there could be race-conditions as well.

Reply:
Agreed. Though race conditions could happen with the fetch-and-filter approach too, at latest when trying to update a resource that has been deleted for reason in the meantime. But yes, what to do with out-of-syncs...

I've added locally more printcolumns so that `k get effectiveschedule` prints the job type, generated schedule and the name and namespace of the first schedule ref. Do you think that's enough for debugging? One would need access to the operator namespace though.
I would not store the name of the effective schedule in the Ready condition's message anymore, only when there was an error.